### ISSUE TITLE
Errors when updating ALIAS records in Constellix

### ISSUE THREAD

**Issue by danielmittelman** [github/octodns#409]
**Description:**
There is a bug in the Constellix provider which does not allow updating (and probably deleting) an ANAME record.

When creating a new ANAME record through OctoDNS it is working perfectly. However when updating a record you see the following error:

```python
requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url: https://api.dns.constellix.com/v1/domains/<...>/records/ALIAS/<...>
```

The issue appears to be that this API call should be made to `.../records/ANAME/...` and not `.../records/ALIAS/...`.

**Steps to reproduce:**
1. Create a zone for Constellix with an ALIAS record, for example:
```yaml
bla:
  ttl: 900
  type: ALIAS
  value: foo.example.com.
```

2. Apply that zone for the first time. Creation of the record will succeed.

3. Change one of the record's parameters, e.g. the TTL. Apply again, the operation will fail.

**Comment by yzguy** [comment]
Looks like this was already fixed up in: https://github.com/github/octodns/pull/401/files#diff-14b5041f95aa0583a8225fbde2b001c8R259, just hasn't been merged yet.

DNSMadeEasy created Constellix, so I originally used the DNSMadeEasy provider as reference. It turned out DNSMadeEasy uses a `/records/<id>` for DELETE, and Constellix uses `/records/<type>/<id>`.

**Comment by ross** [comment]
> Looks like this was already fixed up in: https://github.com/github/octodns/pull/401/files#diff-14b5041f95aa0583a8225fbde2b001c8R259, just hasn't been merged yet.

Yeah. That was a big enough PR with a provider that I don't have access to that I didn't really feel comfortable reviewing it.

**Comment by yzguy** [comment]
No worries, I'll try to check it out this weekend. I closed my Constellix account, but I can open it again.

**Comment by yzguy** [comment]
Just submitted a PR for this fix in #413,

I'm having issues getting back into my Constellix account and am working with their support. I will need more time to look over the other PR as it adds a lot of functionality for pools, which I have not used.
